tests/utils/verification/volumeproperties.go

```diff
@@ -97,6 +100,75 @@ func GetVolumePropertiesDockerCli(volName string, hostname string) string {
 	return op
 }
 
+// GetContainerName returns full container name based on given short name
+func GetContainerName(hostName, shortName string) (string, error) {
+	cmd := dockercli.ListContainers + "--filter name='" + shortName + "' --format '{{.Names}}'"
+	return ssh.InvokeCommand(hostName, cmd)
+}
+
+// GetAllContainers returns all running containers on the give host based on the filtering criteria
+func GetAllContainers(hostName, filter string) (string, error) {
+	cmd := dockercli.ListContainers + "--filter name='" + filter + "' --format '{{.Names}}'"
+	return ssh.InvokeCommand(hostName, cmd)
+}
+
+// VerifyDockerService returns true if the given service is running on
+// the specified VM with specified replicas; false otherwise.
+func VerifyDockerService(hostName, serviceName string, replicas int) bool {
+	log.Printf("Checking docker service [%s] status on VM [%s]\n", serviceName, hostName)
+	out, err := util.ListService(hostName, serviceName)
+	if err != nil {
+		log.Println(err)
+		return false
+	}
+
+	for i := 1; i <= replicas; i++ {
+		if !strings.Contains(out, serviceName+"."+strconv.Itoa(i)) {
+			return false
+		}
+	}
+	return true
+}
+
+// VerifyDockerContainer returns true and the host IP if the containers are running on one of
+// the given docker hosts with specified replicas; otherwise returns false and empty string.
+func VerifyDockerContainer(dockerHosts []string, serviceName string, replicas int) (bool, string) {
```

Comment:
Please refer to swarm_test line 114 - the returned host is needed for the next steps: query the  container name and stop the service *on this specific host*. My first version of this function does not return host actually. But the test failed in CI setup because the container might be spawned in any hosts. So when verifying the running containers, I need to return the host and then run subsequent steps on this specific host.